tmt/steps/prepare/feature.py

```diff
@@ -0,0 +1,197 @@
+import dataclasses
+import re
+from typing import List, Optional
+
+import tmt
+import tmt.base
+import tmt.log
+import tmt.options
+import tmt.steps
+import tmt.steps.prepare
+import tmt.utils
+from tmt.steps.provision import Guest
+from tmt.utils import Command, field
+
+
+class Feature(tmt.utils.Common):
+    """ Base class for feature implementations """
+
+    KEY: str
+
+    def __init__(
+            self,
+            *,
+            parent: tmt.steps.prepare.PreparePlugin,
+            guest: Guest,
+            logger: tmt.log.Logger) -> None:
+        """ Initialize feature data """
+        super().__init__(logger=logger, parent=parent, relative_indent=0)
+        self.guest = guest
+
+    def get_guest_distro(self, guest: Guest, logger: tmt.log.Logger) -> Optional[str]:
```

Comment:
It seems we could make use of `enum` here, in a way similar to package manager detection in https://github.com/teemtee/tmt/blob/main/tmt/steps/provision/__init__.py#L77. No way to make a typo and cause an "unsupported distro" error later by returning `fedora` instead of `Fedora` :)

Reply:
Plus, maybe a "distro family" rather than "distro"?

Also, could you look at `facts.os_release_content['NAME']`? It seems to be very similar to what you are trying to infer here, just using different wording: `Fedora Linux` instead of `Fedora`, `Red Hat Enterprise Linux` instead of `RHEL-N`, where `N` you could get from `os_release_content['VERSION_ID']`, `9.2`. It would make the `/etc/os-release` the source of "what distro are we on?", which is its purpose after all :)

```python
class Distro(enum.Enum):
  FEDORA = enum.auto()
  RHEL_9 = enum.auto()
  ...

os_release = guest.facts.os_release_content

if os_release.get('NAME') == 'Fedora Linux':
  return Distro.FEDORA

if os_release.get('NAME') == 'Red Hat Enterprise Linux' and os_release.get('VERSION_ID', '').startswith('9'):
  return Distro.RHEL_9
...
```